### PR TITLE
[Doc] Add sections about how to enable and configure chunking for producers/consumers (C++ client)

### PR DESCRIPTION
**Pull request by momo-jun**
### Motivation
To enhance the [Pulsar Feature Matrix](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914), add the required document for chunking in the C++ client category.

### Modifications
1. Add a section about how to enable message chunking for producers (C++ client).
image

2. Add a section about how to configure message chunking for consumers (C++ client).
image

3. Change the display order of the content for "consumer" and "producer".

### Documentation
  
- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @BewareMyPower @Anonymitaet @RobertIndie Can you please review this doc PR? Thanks.

File: site2/docs/client-libraries-cpp.md
Comment (BewareMyPower) [inline site2/docs/client-libraries-cpp.md:535]:
```suggestion
You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `setMaxPendingChunkedMessage` and `setAutoOldestChunkedMessageOnQueueFull` parameters. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later. 
```

Comment (momo-jun) [reply site2/docs/client-libraries-cpp.md]: Sharp eyes! Thanks for pointing this out. I've corrected it. PTAL.

**Review by BewareMyPower — APPROVED** [review]

File: site2/docs/client-libraries-cpp.md
Comment (RobertIndie) [inline site2/docs/client-libraries-cpp.md:485]:
```suggestion
You can avoid running a loop with blocking calls with an event based style by using a message listener which is invoked for each message that is received.
```

File: site2/docs/client-libraries-cpp.md
Comment (RobertIndie) [inline site2/docs/client-libraries-cpp.md:535]:
```suggestion
You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `maxPendingChunkedMessage` and `autoOldestChunkedMessageOnQueueFull` parameters. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later. 
```

I think it would be more reasonable to use `maxPendingChunkedMessage` and `autoOldestChunkedMessageOnQueueFull` here. `setMaxPendingChunkedMessage` and `setAutoOldestChunkedMessageOnQueueFull` are the setting methods for these parameters

Comment (RobertIndie) [reply site2/docs/client-libraries-cpp.md:535]: @BewareMyPower I think there is a typo in `autoOldestChunkedMessageOnQueueFull` parameter of the c++ client. It should be `autoAckOldestChunkedMessageOnQueueFull`. PTAL.

Comment (BewareMyPower) [reply site2/docs/client-libraries-cpp.md:535]: Yes. There is a missed `Ack` word. I'll open another PR for it.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
Prevent this PR from merging since I'll open another PR to fix the typo in C++'s API.

File: site2/docs/client-libraries-cpp.md
Comment (BewareMyPower) [inline site2/docs/client-libraries-cpp.md:535]:
```suggestion
You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `setMaxPendingChunkedMessage` and `setAutoAckOldestChunkedMessageOnQueueFull` parameters. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later. 
```

This patch should be applied after #14070 is merged.

File: site2/docs/client-libraries-cpp.md
Comment (BewareMyPower) [inline site2/docs/client-libraries-cpp.md:544]:
```suggestion
ConsumerConfiguration conf;
conf.setAutoAckOldestChunkedMessageOnQueueFull(true);
conf.setMaxPendingChunkedMessage(100);
Consumer consumer;
client.subscribe("my-topic", "my-sub", conf, consumer);
```

This patch should be applied after #14070 is merged.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
This PR should be ready to merge after https://github.com/apache/pulsar/pull/14070 is merged.

/cc @RobertIndie

File: site2/docs/client-libraries-cpp.md
Comment (Anonymitaet) [inline site2/docs/client-libraries-cpp.md:424]:
```suggestion
The message chunking feature is OFF by default. The following is an example of how to enable message chunking when creating a producer.
```

**Comment by Anonymitaet** [comment]
@BewareMyPower https://github.com/apache/pulsar/pull/14070 is merged, can we merge this doc PR?

**Review by BewareMyPower — APPROVED** [review]

**Comment by BewareMyPower** [comment]
@Anonymitaet Yes. I've addressed the related request changes just now.

**Review by RobertIndie — APPROVED** [review]

**Pull request merged**
